lib/doctest.js

```diff
@@ -322,20 +346,25 @@ function wrap$js(test, sourceType) {
          type === 'ImportDeclaration' ||
          type === 'VariableDeclaration' ?
     test[INPUT].value :
+    // TODO: Why are we enqueing two things if it could just be one structure?
     [
       '__doctest.enqueue({',
       '  type: "' + INPUT + '",',
       '  thunk: function() {',
-      '    return ' + test[INPUT].value + ';',
+      '    return (\n' + test[INPUT].value + '\n    );',
       '  }',
       '});'
-    ].concat (test[OUTPUT] == null ? [] : [
+    ].concat (test[OUTPUT].length === 0 ? [] : [
       '__doctest.enqueue({',
       '  type: "' + OUTPUT + '",',
       '  ":": ' + test[OUTPUT].loc.start.line + ',',
       '  "!": ' + test['!'] + ',',
       '  thunk: function() {',
-      '    return ' + test[OUTPUT].value + ';',
+      '    return [\n' + test[OUTPUT].map (function(out) {
+        return '      ' +
+               '{channel: ' + JSON.stringify (out.channel) +
+               ', value: (\n' + out.value + '\n      )}';
```

Comment:
This approach with the newlines is to allow users to append comments to their input and output lines. It does, however, mean that they can no longer append semicolons to their expressions. I'm not sure if that was allowed.

Reply:
If we want to support expressions with semicolons, we'll have to use an IIFE here instead.